src/wallet/walletdb.cpp

```diff
@@ -32,46 +32,38 @@ static std::atomic<unsigned int> nWalletDBUpdateCounter;
 
 bool CWalletDB::WriteName(const std::string& strAddress, const std::string& strName)
 {
-    nWalletDBUpdateCounter++;
```

Comment:
1. (thread 1) Increment the counter.
2. (thread 2) DB gets flushed
3. (thread 1) Actual change happens.
4. (thread 2) Thinks DB hasn't changed, so doesn't flush again.